parm/stage/master_gefs_RT.yaml.j2

```diff
@@ -0,0 +1,90 @@
+#
+# Any number of keys with nested mkdir, copy, and link are permitted
+# Jinja is permitted in this yaml, as long as the keys are:
+# - COMOUT_
+# - DO_ATM, DO_OCN, DO_ICE, etc.
+# For a full list see scripts/exglobal_stage_ic.py
+###################################################################
+
+
+# Set cycle variables
+# ------------------------
+{% set half_window = assim_freq // 2 %}
+{% set half_window_begin = (-half_window | string + "H") | to_timedelta %}
+{% set half_window_end = (half_window | string + "H") | to_timedelta %}
+{% set model_start_date_current_cycle = current_cycle %}
+{% set current_cycle_YMD = current_cycle | to_YMD %}
+{% set current_cycle_HH = current_cycle | strftime("%H") %}
+{% set previous_cycle_YMD = previous_cycle | to_YMD %}
+{% set previous_cycle_HH = previous_cycle | strftime("%H") %}
+{% set p_prefix = previous_cycle | strftime("%Y%m%d.%H0000") %}
+{% set m_prefix = model_start_date_current_cycle | strftime("%Y%m%d.%H0000") %}
+
+# Construct member COM directory lists
+# ------------------------------------
+{% set current_cycle_dict = ({ '${ROTDIR}':ROTDIR,
+                                '${RUN}':RUN,
+                                '${YMD}':current_cycle_YMD,
+                                '${HH}':current_cycle_HH,
+                                '${MEMDIR}': 'mem%03d' | format(ENSMEM) }) %}
+{% set previous_cycle_dict = ({ '${ROTDIR}':ROTDIR,
+                                '${RUN}':RUN,
+                                '${YMD}':previous_cycle_YMD,
+                                '${HH}':previous_cycle_HH,
+                                '${MEMDIR}': 'mem%03d' | format(ENSMEM) }) %}
+
+
+{% set COMOUT_ATMOS_INPUT_MEM = COM_ATMOS_INPUT_TMPL | replace_tmpl(current_cycle_dict) %}
+{% set COMOUT_ATMOS_RESTART_PREV_MEM = COM_ATMOS_RESTART_TMPL | replace_tmpl(previous_cycle_dict) %}
+{% set COMOUT_ATMOS_ANALYSIS_MEM = COM_ATMOS_ANALYSIS_TMPL | replace_tmpl(current_cycle_dict) %}
+{% set COMOUT_ICE_ANALYSIS_MEM = COM_ICE_ANALYSIS_TMPL | replace_tmpl(current_cycle_dict) %}
+{% set COMOUT_ICE_RESTART_PREV_MEM = COM_ICE_RESTART_TMPL | replace_tmpl(previous_cycle_dict) %}
+{% set COMOUT_OCEAN_RESTART_PREV_MEM = COM_OCEAN_RESTART_TMPL | replace_tmpl(previous_cycle_dict) %}
+{% set COMOUT_OCEAN_ANALYSIS_MEM = COM_OCEAN_ANALYSIS_TMPL | replace_tmpl(current_cycle_dict) %}
+{% set COMOUT_MED_RESTART_PREV_MEM = COM_MED_RESTART_TMPL | replace_tmpl(previous_cycle_dict) %}
+{% set COMOUT_WAVE_RESTART_PREV_MEM = COM_WAVE_RESTART_TMPL | replace_tmpl(previous_cycle_dict) %}
+
+###################################################################
+# Initial condition to stage - include components based on switches
+###################################################################
+{% set m_index = (current_cycle_HH | int) // 6 %}
+
+{% if ENSMEM != 0 %}
+# GFS members (80)
+{% set cyc_ranges = (range(1, 31) | list) + (range(21, 51) | list) + (range(41, 71) | list) + (range(61, 81) | list) + (range(1, 11) | list) %}
```

Comment:
I think this is returning a single list of `[1,..31, 20,..50, ....]`, but what you want is a list of lists `[[1..30], [21..50], [41..70], [60..80, 1..10]]`.  To do that, just add a set of brackets and well-spaced commas:
```jinja
{% set cyc_ranges = [(range(1, 31) | list), (range(21, 51) | list), (range(41, 71) | list), (range(61, 81) | list) + (range(1, 11) | list)] %}
```